Assets/HoloToolkit/Input/Scripts/Focus/FocusManager.cs

```diff
@@ -282,165 +211,77 @@ public void RegisterPointer(IPointingSource pointingSource)
             {
-                if (pointers[i].PointingSource.OwnsInput(eventData))
+                if (focusers[i].OwnsInput(eventData))
                 {
-                    pointingSource = pointers[i].PointingSource;
+                    focuser = focusers[i];
                     return true;
                 }
             }
 
-            pointingSource = null;
+            focuser = null;
             return false;
         }
 
-        public FocusDetails GetFocusDetails(IPointingSource pointingSource)
-        {
-            PointerData pointerData;
-            FocusDetails details = default(FocusDetails);
-
-            if (GetPointerData(pointingSource, out pointerData))
-            {
-                details = pointerData.End;
-            }
-
-            return details;
-        }
-
-        public GameObject GetFocusedObject(IPointingSource pointingSource)
-        {
-            PointerData pointerData;
-            GameObject focusedObject = null;
-
-            if (GetPointerData(pointingSource, out pointerData))
-            {
-                focusedObject = pointerData.End.Object;
-            }
+        public delegate void FocusMethodInfo(InputManager.FocusEvent focusedObject);
+        public event FocusMethodInfo FocusEnteredInfo;
+        public event FocusMethodInfo FocusExitedInfo;
 
-            return focusedObject;
-        }
+        private PointerInputEventData pointerInputEventData;
 
         /// <summary>
-        /// Checks if exactly one pointer is registered and returns it if so.
+        /// Called when a focus target without focus gains focus from one or more focusers
         /// </summary>
-        /// <returns>The registered pointer if exactly one is registered, null otherwise.</returns>
-        public bool TryGetSinglePointer(out IPointingSource pointingSource)
-        {
-            if (pointers.Count == 1)
-            {
-                pointingSource = pointers[0].PointingSource;
-                return true;
-            }
-
-            pointingSource = null;
-            return false;
-        }
-
-        public delegate void FocusEnteredMethod(GameObject focusedObject);
-        public event FocusEnteredMethod FocusEntered;
+        public delegate void FocusGainedMethod(IFocusTarget focusedObject);
+        public event FocusGainedMethod FocusGained;
 
-        public delegate void FocusExitedMethod(GameObject unfocusedObject);
-        public event FocusExitedMethod FocusExited;
-
-        public delegate void PointerSpecificFocusChangedMethod(IPointingSource pointer, GameObject oldFocusedObject, GameObject newFocusedObject);
-        public event PointerSpecificFocusChangedMethod PointerSpecificFocusChanged;
+        /// <summary>
+        /// Called when a focus target with focus loses focus from ALL focusers
+        /// </summary>
+        public delegate void FocusLostMethod(IFocusTarget unfocusedObject);
+        public event FocusLostMethod FocusLost;
```

Comment:
There's no reason to have these events if we're already raising them through the InputManager

Reply:
These were moved from InputManager to FocusManager for clarity.